modules/carto/src/basemap.ts

```diff
@@ -1,4 +1,91 @@
-const baseUrl = 'https://basemaps.cartocdn.com/gl/{basemap}-gl-style/style.json';
+export const cartoBasemapsBaseUrl = 'https://basemaps.cartocdn.com/gl';
+
+const baseUrl = `${cartoBasemapsBaseUrl}/{basemap}-gl-style/style.json`;
+
+export const cartoStyleLayerGroups = [
+  {
+    slug: 'label',
+    filter: ({id}: {id: string}) =>
+      Boolean(
+        id.match(/(?=(label|_label|place-|place_|poi-|poi_|watername_|roadname_|housenumber))/)
+      ),
+    defaultVisibility: true
+  },
+  {
+    slug: 'road',
+    filter: ({id}: {id: string}) =>
+      Boolean(id.match(/(?=(road|railway|tunnel|street|bridge))(?!.*label)/)),
+    defaultVisibility: true
+  },
+  {
+    slug: 'border',
+    filter: ({id}: {id: string}) => Boolean(id.match(/border|boundaries|boundary_/)),
+    defaultVisibility: false
+  },
+  {
+    slug: 'building',
+    filter: ({id}: {id: string}) => Boolean(id.match(/building/)),
+    defaultVisibility: true
+  },
+  {
+    slug: 'water',
+    filter: ({id}: {id: string}) => Boolean(id.match(/(?=(water|stream|ferry))/)),
+    defaultVisibility: true
+  },
+  {
+    slug: 'land',
+    filter: ({id}: {id: string}) =>
+      Boolean(id.match(/(?=(parks|landcover|industrial|sand|hillshade|park_))/)),
+    defaultVisibility: true
+  }
+];
+
+export function applyCartoLayerGroupFilters(style, visibleLayerGroups: Record<string, boolean>) {
```

Comment:
Update `Record<string, boolean>` to extract valid keys from `cartoStyleLayerGroups`?